### ISSUE TITLE
Bug: Chrome logger does not work.

### ISSUE THREAD

**Issue by dafriend** [codeigniter4/CodeIgniter4#2616]
**Describe the bug**
Plain and simple, doesn't work. I think the framework changed significantly since the code was first written. The event that was to send the headers doesn't exist any longer.

I discovered this some time ago and devised a fix but never made a PR on it.

Feel free to assign this one to me. It might take a little while to get time for it. I mostly need to devise some unit tests.

**Comment by MGatner** [comment]
Appreciate that you offered to look into it. I don’t use Chrome so I’m pretty unfamiliar with the structure but... any ideas why the tests are passing even if this has been broken for a while?

**Comment by dafriend** [comment]
I knew at one point, but have forgotten now.

**Comment by dafriend** [comment]
I refreshed my memory about the issue because I know y'all are curious why it passes the unit test but fails in the real world.

The unit test method calls ``ChromeLoggerHandler::sendLogs()`` which puts the required header into a response object. The test then checks the current response object for the head. It's there all right. Assertion true and we're in the green.

However.  here is how ``ChromeLoggerHandler`` is set up to call ``sendLogs()``

`Events::on('post_controller', [$this, 'sendLogs'], EVENT_PRIORITY_HIGH);`

Except there isn't an event named _'post_controller'_, and a non-existent event won't be triggered. Obviously.

**Comment by MGatner** [comment]
You are so right! Good catch. I’d say just try switching to `post_controller_constructor` and see if that magically fixes it?

**Comment by dafriend** [comment]
I actually have... er, that is, had a solution. I've made a mess of my local repo and cannot find the final working commit. It's in there somewhere. 

It would be nice to find since I tackled the @todo and worked out some fancy object formatting which users should appreciate. I also added logic so that the ChromeLogger cannot be instantiated in the production environment. We would be quite negligent to allow that to happen.

I'm on it though.
